inc/plugins/myalerts.php

```diff
@@ -1206,6 +1207,56 @@ function myalerts_alert_rated_threadauthor()
     }
 }
 
+$plugins->add_hook(
+	'polls_vote_end',
+	'myalerts_alert_voted_threadauthor'
+);
+function myalerts_alert_voted_threadauthor()
+{
+	global $mybb, $db, $poll;
+
+    if (!isset($mybb->user['uid']) || $mybb->user['uid'] < 1) {
+        return;
+    }
+
+    myalerts_create_instances();
+
+    $alertTypeManager = MybbStuff_MyAlerts_AlertTypeManager::getInstance();
+
+    if (is_null($alertTypeManager) || $alertTypeManager === null) {
+        global $cache;
+
+        $alertTypeManager = MybbStuff_MyAlerts_AlertTypeManager::createInstance($db, $cache);
+    }
+
+    $alertType = $alertTypeManager->getByCode('voted_threadauthor');
+
+    if ($alertType != null && $alertType->getEnabled()) {
+        $thread = get_thread($poll['tid']);
+
+        if ($thread['uid'] != $mybb->user['uid']) {
+            $forumPerms = forum_permissions($thread['fid'], $thread['uid']);
+
+            // Check forum permissions
+            if ($forumPerms['canview'] != 0 || $forumPerms['canviewthreads'] != 0) {
+                $alert = new MybbStuff_MyAlerts_Entity_Alert(
+                    $thread['uid'],
+                    $alertType,
+                    (int) $tid
+                );
+                $alert->setExtraDetails(
+                    array(
+                        'tid'       => $poll['tid'],
+                        't_subject' => $thread['subject'],
+                    )
+                );
+
+                MybbStuff_MyAlerts_AlertManager::getInstance()->addAlert($alert);
```

Comment:
Again, you need to check that `getInstance()` here doesn't return false either and create the instance if needed.